### PR TITLE
FIX-#4649: try to read csv/fwf data with the first line after header in the case of `pandas.errors.ParserError`

### LINKED ISSUE #4649: `read_csv` throws errors in case using `quoting=csv.QUOTE_NONE` parameter
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Windows 11 
- **Modin version** (`modin.__version__`): 35ae83481aed58c8a8abfc97b900c494a5d47f21
- **Python version**: 3.8.13
- **Code we can use to reproduce**:

```python
import csv

import modin.pandas as pd
#import pandas as pd
import modin.config as cfg
cfg.Engine.put("ray")

filename = "test_data.csv"
df = pd.read_csv(filename, quoting=csv.QUOTE_NONE)
print(df)

```

test_data.csv :
```csv
a,b
0,"b0,b1"
1,"b2,b3"
2,c
3,d
4,"m,x"
5,"p,k"

```

### Describe the problem

Modin throws errors in case of using `read_csv` with `quoting=csv.QUOTE_NONE` parameter but pandas doesn't.

### Source code / logs

Pandas output:

```shell
     a    b
0  "b0  b1"
1  "b2  b3"
2    c  NaN
3    d  NaN
4   "m   x"
5   "p   k"
```

Modin output:

```shell
  File "C:\Users\79049\Documents\projects\modin\modin\core\execution\ray\common\task_wrapper.py", line 42, in _deploy_ray_func
    return func(**args)
  File "C:\Users\79049\Documents\projects\modin\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\Users\79049\Documents\projects\modin\modin\core\storage_formats\pandas\parsers.py", line 298, in parse
    return PandasParser.generic_parse(fname, **kwargs)
  File "C:\Users\79049\Documents\projects\modin\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\Users\79049\Documents\projects\modin\modin\core\storage_formats\pandas\parsers.py", line 197, in generic_parse
    pandas_df = callback(BytesIO(to_read), **kwargs)
  File "C:\Users\79049\Documents\projects\modin\modin\logging\logger_decorator.py", line 128, in run_and_log
    return obj(*args, **kwargs)
  File "C:\Users\79049\Documents\projects\modin\modin\core\io\text\csv_dispatcher.py", line 40, in read_callback
    return pandas.read_csv(*args, **kwargs)
  File "C:\Users\79049\.conda\envs\modin\lib\site-packages\pandas\util\_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "C:\Users\79049\.conda\envs\modin\lib\site-packages\pandas\io\parsers\readers.py", line 680, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "C:\Users\79049\.conda\envs\modin\lib\site-packages\pandas\io\parsers\readers.py", line 581, in _read
    return parser.read(nrows)
  File "C:\Users\79049\.conda\envs\modin\lib\site-packages\pandas\io\parsers\readers.py", line 1255, in read
    index, columns, col_dict = self._engine.read(nrows)
  File "C:\Users\79049\.conda\envs\modin\lib\site-packages\pandas\io\parsers\c_parser_wrapper.py", line 225, in read
    chunks = self._reader.read_low_memory(nrows)
  File "pandas\_libs\parsers.pyx", line 805, in pandas._libs.parsers.TextReader.read_low_memory
  File "pandas\_libs\parsers.pyx", line 861, in pandas._libs.parsers.TextReader._read_rows
  File "pandas\_libs\parsers.pyx", line 847, in pandas._libs.parsers.TextReader._tokenize_rows
  File "pandas\_libs\parsers.pyx", line 1960, in pandas._libs.parsers.raise_parser_error
pandas.errors.ParserError: Error tokenizing data. C error: Expected 2 fields in line 4, saw 3
```

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4649 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/storage_formats/pandas/parsers.py [hunk 1]
```diff
@@ -200,13 +204,30 @@ def generic_parse(fname, **kwargs):
...
+
+        try:
+            pandas_df = callback(BytesIO(header_with_to_read), **kwargs)
+        except pandas.errors.ParserError:
+            # looks like pandas determines the number of fields by the first line
+            # after the header, let's try like this and discard it after reading;
+            # as a test of the approach's operability, it's better to use it in some
+            # cases, not in all
+            if kwargs["quoting"] == csv.QUOTE_NONE and first_line:
+                pandas_df = callback(BytesIO(header + first_line + to_read), **kwargs)
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/parsers.py:225]: doesn't this tell us that `start` and `end` was computed incorrectly? is there a way to compute them properly at the beginning?

File: modin/core/storage_formats/pandas/parsers.py [hunk 2]
```diff
@@ -200,13 +204,30 @@ def generic_parse(fname, **kwargs):
...
+                # drop the first line
+                pandas_df = pandas_df.iloc[1:]
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py:227]: The header is the information that should form the columns of the dataframe. In this case, the first line of the dataframe is the first line of the file after the header.

File: modin/core/storage_formats/pandas/parsers.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/storage_formats/pandas/parsers.py:225]: Let's see how we split the test file:
```python
# first partition
a,b
0,"b0,b1"
1,"b2,b3"
# second partition
2,c
3,d
# third partition
4,"m,x"
5,"p,k"
```
It seems to me that the split is correct, just not good in terms of how pandas processes rows where there are not all fields.

**Comment by anmyachev** [comment]
> Unfortunately, this is not a full fix... imagine this line:
> 
> ```python
> pd.read_csv(io.BytesIO(b'a,b\nc,d,e\n'), header=None, on_bad_lines=lambda x:[],engine='python')`
> ```
> 
> this produces
> 
> ```
>       0     1
> 0     a     b
> 1  None  None
> ```
> 
> Now imagine the split happening _exactly_ between short rows and long rows... I do recall us having an issue caused by that, but I cannot find it easily, maybe @YarShev remembers?..

You're right.

**Pull request closed without merging**